DFC.EventGridSubscriptions.ApiFunction/StartUp/FunctionStartupExtension.cs

```diff
@@ -0,0 +1,52 @@
+﻿using DFC.EventGridSubscriptions.ApiFunction.StartUp;
+using DFC.EventGridSubscriptions.Data;
+using DFC.EventGridSubscriptions.Services;
+using DFC.EventGridSubscriptions.Services.Interface;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Azure.Management.EventGrid;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Rest;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text;
+
+[assembly: FunctionsStartup(typeof(FunctionStartupExtension))]
+
+namespace DFC.EventGridSubscriptions.ApiFunction.StartUp
+{
+    /// <summary>
+    /// The function startup extension.
+    /// </summary>
+    [ExcludeFromCodeCoverage]
+    public class FunctionStartupExtension : FunctionsStartup
+    {
+        /// <inheritdoc/>
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            var config = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+                .AddJsonFile("appsettings.development.json", optional: true, reloadOnChange: true)
```

Comment:
Best not to hard-code loading of the development settings, in case they get copied to the output directory.
Better to load the relevant file for the env, or only load the development version if it is dev.

Reply:
Good spot, thank you